druntime/src/core/thread/osthread.d

```diff
@@ -2278,6 +2278,29 @@ else version (Posix)
         __gshared sem_t suspendCount;
 
 
+        extern (C) void thread_preSuspend( void* sp ) nothrow {
+            // NOTE: Since registers are being pushed and popped from the
+            //       stack, any other stack data used by this function should
+            //       be gone before the stack cleanup code is called below.
+            Thread obj = Thread.getThis();
+            assert(obj !is null);
+
+            if ( !obj.m_lock )
+            {
+                obj.m_curr.tstack = getStackTop();
```

Comment:
Why is getStackTop called here? doesn't it come in from the parameter?

Reply:
Right.